### PR TITLE
Refactor RequestContext.cs according to the latest technical design

### PR DESCRIPTION
**Pull request by laura-rodriguez**
* Mark as obsolete the previous strategy
* Create a RequestContext object and pass it to the Interact method
* Allow methods that internally call interact to receive a request context
* Update readme

### CODE REVIEW

**Review by arvindkrishnakumar-okta — APPROVED** [review]
lgtm

**Review by bryanapellanes-okta — APPROVED** [review]
👍

File: README.md [hunk 1]
```diff
@@ -71,20 +71,22 @@ var client = new IdxClient(new IdxConfiguration()
...
-deviceContext.UseHeader(RequestHeaders.XDeviceToken, "deviceToken"); // Only for confidential clients
-deviceContext.UseHeader(RequestHeaders.UserAgent, "MyUserAgent");
-deviceContext.UseHeader(RequestHeaders.XOktaUserAgentExtended, "MyXUserAgent");
-deviceContext.UseHeader(RequestHeaders.XForwardedFor, "MyForwardedFor");
+var requestContext = new RequestContext
+                    {
+                        XDeviceToken = "deviceToken",
+                        XOktaUserAgentExtended = "MyXUserAgent",
+                        XForwardedFor = "MyForwardedFor",
+                    };
```
Comment (bdemers) [inline README.md:84]: Suggestion (not a blocker)

From a developer's perspective, these names leak the actual implementation.  For example, this constructor needs the User-Agent string, but the developer likely won't know that the user-agent actually maps back to `X-Okta-User-Agent-Extended` header, similar to `X-Forwarded-For`, which is an IP address.

The actual implementation could change to move these into the POST body with different attribute names.

Consider changing the names to, `DeviceToken`, `UserAgent`, and `IpAddress`:

```csharp
var requestContext = new RequestContext
    {
        DeviceToken = "deviceToken",
        UserAgent = "MyUserAgent",
        IpAddress = "MyIpAddress",
    };
```

**Review by bdemers — COMMENTED** [review]
Looks good! added a suggestion that may help the reader/developer understand the purpose of the RequestContext. (not blocking)

File: README.md (hunk 1 above)
Comment (laura-rodriguez) [reply README.md]: Thanks for the suggestion Demers! I removed the X prefix from DT and UA but we're keeping the XFF name to be consistent with classic.

**Pull request merged**